_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp

```diff
@@ -1284,11 +1284,20 @@ mfxStatus VideoDECODEVP8_HW::DecodeFrameHeader(mfxBitstream *in)
         while (++i < 2);
     }
 
+// On Android O we use old version of driver and should use special code for 1st partition size computation (for count == 8)
+#if (MFX_ANDROID_VERSION >= MFX_P)
     // Header info consumed bits
     m_frame_info.entropyDecSize = m_boolDecoder[VP8_FIRST_PARTITION].pos() * 8 - 3*8 - m_boolDecoder[VP8_FIRST_PARTITION].bitcount();
 
     // Subtract completely consumed bytes + current byte. Current is completely consumed if bitcount is 8.
     m_frame_info.firstPartitionSize = first_partition_size - ((m_frame_info.entropyDecSize + 7) >> 3);
+#else
+    // Header info consumed bits
+    m_frame_info.entropyDecSize = m_boolDecoder[VP8_FIRST_PARTITION].pos() * 8 - 16 - m_boolDecoder[VP8_FIRST_PARTITION].bitcount();
```

Comment:
why not 2*8 like above (line 1290)?
just looks better, not required:)

Reply:
Agree, that it would be better, but current patch recovers original code for calculating macroblock for VP8D without any changes, and original code  - "16"